### ISSUE TITLE
Has this been tested with react 0.13.0-rc2?

### ISSUE THREAD

**Issue by terebentina** [gaearon/react-hot-loader#95]
I upgraded my app yesterday to react 0.13 rc2 and operated the changes required by it (got rid of classSet / changed cloneWithProps to the new cloneElement / React.findDOMNode instead of component.getDOMNode()). 

I now just noticed that changes to my jsx are not reloaded (neither hot nor auto page refresh). They only show up after I manually refresh the page.

I do get these from the dev-server after every change but no jsx changes (scss changes are hot reloaded properly):

[WDS] App updated. Recompiling...
[WDS] App updated. Recompiling...
[WDS] App updated. Recompiling...
[WDS] App updated. Recompiling...
[WDS] App hot update...
[WDS] App hot update...
[HMR] Checking for updates on the server...
XHR finished loading: GET "http://localhost:8080/dist/512f6d3dc9638c57e6e6.hot-update.json".
[HMR] Updated modules:
[HMR]  - 268
[HMR] App is up to date.

**Comment by gaearon** [comment]
I'll take another look, thanks.

**Comment by gaearon** [comment]
It works with boilerplate, I tried React 0.13 RC2 in a branch: https://github.com/gaearon/react-hot-boilerplate/tree/react-0.13

Maybe you got a duplicate React installation by mistake? This may happen if some library specifies React in `dependencies` instead of `peerDependencies`, and when you update to a new version of React, it keeps the old version in its own `node_modules`.

**Comment by terebentina** [comment]
Hmm, no, I checked...could it be from the fact that I have 2 entry points? A vendors and an app?
This is the only significant difference I see from the boilerplate. And the fact that I have a real server started, besides webpack-dev-server.
I was starting webpack-dev-server with --inline --hot but now I removed the command line args and added them in webpack config, just for the app entry.
I also added the webpack.HotModuleReplacementPlugin plugin and config.devServer = { hot: true }
Still, it behaves the same.

**Comment by gaearon** [comment]
Can you check https://github.com/gaearon/react-hot-loader/blob/master/docs/Troubleshooting.md page if any suggestions could apply to your case?

**Comment by terebentina** [comment]
Everything is fine according to this page. However, it lead me to find the problem:
I had this code in my index.js even though react was a npm package in node_modules:
```javascript
if (module.hot) {
  require('react-hot-loader/Injection').RootInstanceProvider.injectProvider({
    getRootInstances: function () {
      // Help React Hot Loader figure out the root component instances on the page:
      return [rootInstance];
    }
  });
}
```
When I removed the code, hot reload started working.
You might want to mention this in the troubleshooting page specifically?
Anyway, thanks for your help, it's much appreciated.

**Comment by gaearon** [comment]
>I had this code in my index.js even though react was a npm package in node_modules:

Where did it come from? It should only be used with external React. Regardless of this, *this code should actually work too* with the current version, `react-hot-loader@1.2.1`. It was fixed in 1.2.0.
